compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6412,8 +6412,16 @@ bool directMemoryStoreHelper(TR::CodeGenerator* cg, TR::Node* storeNode)
                TR::DebugCounter::incStaticDebugCounter(cg->comp(), "z/optimization/directMemoryStore/indirect");
 
                // Force the memory references to not use an index register because MVC is an SS instruction
+               // After generating a memory reference, Enforce it to generate LA instruction.
+               // This will avoid scenarios when we have common base/index between destination and source
+               // And when generating the source memory reference, it clobber evaluates one of the node shared between 
+               // target memory reference as well.
+               // TODO : Fix this. It should not clobber evaluate the node in first place. Workaround just suppresses the bug.
```

Comment:
I have opened up issue against this and linked this PR with that with rmoving TODO from the comment.